### PR TITLE
Fixtures

### PR DESCRIPTION
**Pull request by rogiervandergeer**
Replace all inheritance structures in the tests with fixtures

### CODE REVIEW

File: tests/test_logging.py [hunk 1]
```diff
@@ -10,23 +7,22 @@
...
+        with open(log_file, "r") as f:
             assert len(f.readlines()) == len(simple_chromosomes)
         # we should see that a file was created with an appropriate number of rows
         pop.log()
-        with open(filepath, "r") as f:
+        with open(log_file, "r") as f:
             assert len(f.readlines()) == (2*len(simple_chromosomes))
-        os.remove(filepath)
 
-    def test_baselogger_can_accept_kwargs(self, simple_chromosomes, simple_evaluation_function):
```
Comment (koaning) [inline tests/test_logging.py]: this is much better this way.

File: tests/test_population.py [hunk 2]
```diff
@@ -78,122 +59,121 @@ def raise_function(_):
...
-            pop3.survive()
+            simple_population.survive()
 
 
-class TestPopulationBreed(TestPopulation):
+class TestPopulationBreed:
 
-    def test_breed_amount_works(self):
-        pop1 = Population(chromosomes=self.chromosomes, eval_function=self.eval_func)
+    def test_breed_amount_works(self, simple_chromosomes, simple_evaluation_function):
```
Comment (koaning) [inline tests/test_population.py]: it feels a bit magic-y if you've never used pytest like this before, but it seems to work alright.

**Comment by koaning** [comment]
I've approved, but we need to fix the conflicting files. I don't have time for this tonight, maybe tomorrow.

**Comment by koaning** [comment]
Somethings strange is happening here. Let's merge this tomorrow and push the button.

**Comment by rogiervandergeer** [comment]
@koaning I think this can be merged

**Pull request merged**